pycbc/io/live.py

```diff
@@ -56,35 +56,39 @@ def __init__(self, ifos, coinc_results, **kwargs):
             Dictionary providing arguments to be used in source probability
             estimation with `pycbc/mchirp_area.py`.
         """
-        self.template_id = coinc_results[f'foreground/{ifos[0]}/template_id']
+        self.template_id = \
+            coinc_results[f'foreground/{coinc_ifos[0]}/template_id']
         self.coinc_results = coinc_results
-        self.ifos = ifos
         self.basename = None
 
-        # remember if this should be marked as HWINJ
+        # Remember if this should be marked as HWINJ
         self.is_hardware_injection = ('HWINJ' in coinc_results
                                       and coinc_results['HWINJ'])
 
-        # Check if we need to apply a time offset (this may be permerger)
+        # Check if we need to apply a time offset for pre-merger triggers
         self.time_offset = 0
-        rtoff = f'foreground/{ifos[0]}/time_offset'
+        rtoff = f'foreground/{coinc_ifos[0]}/time_offset'
         if rtoff in coinc_results:
             self.time_offset = coinc_results[rtoff]
 
+        # Process exactly those ifos for which SNR is calculated
         if 'followup_data' in kwargs:
             fud = kwargs['followup_data']
             assert len({fud[ifo]['snr_series'].delta_t for ifo in fud}) == 1, \
                     "delta_t for all ifos do not match"
             self.snr_series = {ifo: fud[ifo]['snr_series'] for ifo in fud}
-            usable_ifos = fud.keys()
-            followup_ifos = list(set(usable_ifos) - set(ifos))
+            snr_ifos = fud.keys()
 
             for ifo in self.snr_series:
                 self.snr_series[ifo].start_time += self.time_offset
         else:
             self.snr_series = None
-            usable_ifos = ifos
-            followup_ifos = []
+            snr_ifos = coinc_ifos
+
+        # Set merger time to the mean of ifo peaks - to be used for file naming
+        self.merger_time = numpy.mean(
+                    [coinc_results[f'foreground/{i}/end_time']
+                     for i in snr_ifos]) + self.time_offset
```

Comment:
I know this is probably not what the old code was doing, but the average should be taken over `coinc_ifos` instead, no? Otherwise we are including potentially unreliable time estimates in the average.

Reply:
If you mean the O3 code, indeed it took all ifos with a SNR time series, and those all had an identified peak/trigger time. Is the point here that the peak time was 'unreliable' if the SNR was too low (eg GW170817 in V1)?
We can discuss whether this is enough of a problem that we want to make a change wrt O3 behaviour, but that would be beyond the scope of this PR. I have tried to only reproduce (and rationalise!) O3 behaviour here concerning all candidate info except pastro. I think it might confuse reviewers if we choose to change the definition of the event time in the same PR as the pastro change.